### PR TITLE
Updating the coding standards

### PR DESCRIPTION
**Pull request by jessicafalk**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
updated the coding standards

#### Tests
none needed
#### Documentation
How is this documented (for example: release note, upgrade guide, feature page, in-code documentation)?
#### Primary reviewers
If your change will take a long time to review, you can name at most two primary reviewers who are ultimately responsible for reviewing this request. @ mention them.

### CODE REVIEW

File: docs/contributions/unity-gdk-coding-standards.md [hunk 1]
```diff
@@ -69,26 +67,22 @@ If you don't have ReSharper, or want to format from the command line, run `ci/li
 
 ## Tests
 
-* Name test fixtures as `<class>Test`
+* Name test fixtures as `<class>Tests`
 * Name test methods as `<method>_should_<action>_when_<conditions>`
 
 ## Unity specific
 
-* We are using the experimental support for `.NET 4.6` in the Unity GDK (mandated by use of the Unity ECS). This equates to `C# 6` so newer language features are supported.
```
Comment (samiwh) [inline docs/contributions/unity-gdk-coding-standards.md]: Could also specify that it's the non-experimental .NET 4.x support. Which is what Unity calls it. That is C# 7 ish though.

File: docs/contributions/unity-gdk-coding-standards.md [hunk 2]
```diff
@@ -69,26 +67,22 @@ If you don't have ReSharper, or want to format from the command line, run `ci/li
...
+* We are using the experimental support for `C# 7` in the SpatialOS GDK for Unity (mandated by use of the Unity ECS).
 * Implement logic in a `ComponentSystem` instead of a `MonoBehaviour` wherever possible.
-* Each call to `Debug.Log` takes ~11ms (even in built players!), so use it sparingly ([relevant thread in Unity Answers](https://answers.unity.com/questions/126315/debuglog-in-build.html)).
```
Comment (samiwh) [inline docs/contributions/unity-gdk-coding-standards.md]: 👍

File: docs/contributions/unity-gdk-coding-standards.md [hunk 3]
```diff
@@ -69,26 +67,22 @@ If you don't have ReSharper, or want to format from the command line, run `ci/li
...
 * Make sure you remove all `Debug.Log` statements before opening a PR.
 * Avoid running `foreach` over an `IEnumerable<T>` because it allocates excessively. See [this StackOverflow question](https://stackoverflow.com/questions/19689328/why-ienumerable-slow-and-list-is-fast) for an explanation of why `IEnumerable<T>` allocates.
 * Be aware of the possible allocations [when using collections](https://jacksondunstan.com/articles/3805) and avoid doing so where the volume would impact performance.
-* When using structs, implement a custom hash code function and the `IEquatable<>` interface. If you do **not** do this, performance will worsen and you will see unexpected allocations; for example, when using the struct in collections.
-* Avoid using enums as dictionary keys. This leads to extra allocations due to boxing in the Mono runtime.
-    * This appears to be true as of 2014, see [here](https://stackoverflow.com/questions/26280788/dictionary-enum-key-performance).
+* When using structs as keys in dictionaries or sets, ensure to implement a custom hash code function and the `IEquatable<>` interface to avoid a performance drop.
```
Comment (samiwh) [inline docs/contributions/unity-gdk-coding-standards.md]: Also implement IEquatable when that struct is used in comparisons.

File: docs/contributions/unity-gdk-coding-standards.md [hunk 4]
```diff
@@ -69,26 +67,22 @@ If you don't have ReSharper, or want to format from the command line, run `ci/li
...
     * Write all of them in a forward-compatible way.
-    * Write `if (!(UNITY_5_0 | UNITY_5_1))` instead of `if (UNITY_5_2)` if you want to specify `5.2` or newer.
-    * Be careful with defines such as `UNITY_5_3_OR_NEWER` - they might not be available in all versions of Unity 5.3.
-* Use `== null` when testing `GameObject`s and `MonoBehaviour`s even though they overload `operator!`.
+    * Write `if (!(UNITY_2018_0 | UNITY_2018_1))` instead of `if (UNITY_2018_2)` if you want to specify `2018.2` or newer.
+    * Be careful with defines such as `UNITY_2018_1_OR_NEWER` - they might not be available in all versions of Unity 2018.1.
+* Use `== null` when testing `GameObject`s and `MonoBehaviour`s even though they overload the `operator!`.
 * Avoid using custom threads. They complicate things and can easily cause crashes.
-* Avoid C# events. They are not efficient and allocate.
 * If an error occurs from which it is impossible to recover, throw exceptions instead of using `Debug.LogError`.
```
Comment (samiwh) [inline docs/contributions/unity-gdk-coding-standards.md]: I'd get rid of this line too.

File: docs/contributions/unity-gdk-coding-standards.md [hunk 5]
```diff
@@ -65,30 +63,25 @@ If you don't have ReSharper, or want to format from the command line, run `ci/li
...
 ## Tests
 
-* Name test fixtures as `<class>Test`
+* Name test fixtures as `<class>Tests`
 * Name test methods as `<method>_should_<action>_when_<conditions>`
 
 ## Unity specific
 
-* We are using the experimental support for `.NET 4.6` in the Unity GDK (mandated by use of the Unity ECS). This equates to `C# 6` so newer language features are supported.
+* We are using the `.NET 4.x` support in the SpatialOS GDK for Unity (mandated by use of the Unity ECS). This equates to `C# 7` so never language features are supported.
```
Comment (jared-improbable) [inline docs/contributions/unity-gdk-coding-standards.md]: *newer

File: docs/contributions/unity-gdk-coding-standards.md [hunk 6]
```diff
@@ -65,30 +63,25 @@ If you don't have ReSharper, or want to format from the command line, run `ci/li
...
 * Make sure you remove all `Debug.Log` statements before opening a PR.
 * Avoid running `foreach` over an `IEnumerable<T>` because it allocates excessively. See [this StackOverflow question](https://stackoverflow.com/questions/19689328/why-ienumerable-slow-and-list-is-fast) for an explanation of why `IEnumerable<T>` allocates.
 * Be aware of the possible allocations [when using collections](https://jacksondunstan.com/articles/3805) and avoid doing so where the volume would impact performance.
-* When using structs, implement a custom hash code function and the `IEquatable<>` interface. If you do **not** do this, performance will worsen and you will see unexpected allocations; for example, when using the struct in collections.
-* Avoid using enums as dictionary keys. This leads to extra allocations due to boxing in the Mono runtime.
-    * This appears to be true as of 2014, see [here](https://stackoverflow.com/questions/26280788/dictionary-enum-key-performance).
+* When using structs as keys in dictionaries or sets or in comparison, ensure to implement a custom hash code function and the `IEquatable<>` interface to avoid a performance drop.
```
Comment (jared-improbable) [inline docs/contributions/unity-gdk-coding-standards.md]: "dictionaries, sets or in comparisons"

File: docs/contributions/unity-gdk-coding-standards.md (hunk 5 above)
Comment (jessicafalk) [reply docs/contributions/unity-gdk-coding-standards.md]: that was a bad typo 😂

**Pull request merged**